src/array/kernel.cc

```diff
@@ -333,35 +350,41 @@ DGL_REGISTER_GLOBAL("sparse._CAPI_DGLKernelSpMMHetero")
     List<Value> list_U = args[3];
     List<Value> list_E = args[4];
     List<Value> list_V = args[5];
-    NDArray ArgU = args[6];
-    NDArray ArgE = args[7];
+    List<Value> list_ArgU = args[6];
+    List<Value> list_ArgE = args[7];
+    List<Value> list_ArgU_ntype = args[8];
+    List<Value> list_ArgE_etype = args[9];
     std::vector<NDArray> U_vec;
     std::vector<NDArray> V_vec;
     std::vector<NDArray> E_vec;
+    std::vector<std::vector<NDArray>> Arg_vec;  // ArgU + ArgE
     U_vec.reserve(list_U.size());
     V_vec.reserve(list_V.size());
     E_vec.reserve(list_E.size());
-    for (Value val : list_U) {
-      U_vec.push_back(val->data);
-    }
-    for (Value val : list_V) {
-      V_vec.push_back(val->data);
-    }
-    for (Value val : list_E) {
-      E_vec.push_back(val->data);
-    }
+
+    for (Value val : list_U) U_vec.push_back(val->data);
+    for (Value val : list_V) V_vec.push_back(val->data);
+    for (Value val : list_E) E_vec.push_back(val->data);
```

Comment:
There should be a `ListValueToVector` function doing the same thing.

Reply:
Yes, found it. Replaced.